### PR TITLE
issue #172: Redirect old archive URLs to the new location

### PR DESCRIPTION
**Pull request by cmpilato**

### CODE REVIEW

**Comment by cmpilato** [comment]
I'm not a PHP developer, but I _suspect_ there's some URI-encoding that needs to happen to the `cat` CGI parameter before it gets stuffed onto the end of the redirect URL.  Happy to make whatever changes are suggested.

File: PodcastGenerator/index.php [hunk 1]
```diff
@@ -9,6 +9,17 @@
...
 session_start();
 require 'core/include.php';
+
+// Backwards compatibility: Redirect pre-3.0 archive pages to
+// categories.php.
+if(isset($_GET['p'])) {
+    if($_GET['p'] == 'archive') {
+        $redirect_url = $config['url'] . 'categories.php?cat=' . $_GET['cat'];
+        header('Location: ' . $redirect_url, TRUE, 301);
+        return;
```
Comment (emilengler) [inline PodcastGenerator/index.php]: Could you change this to `die();`

File: PodcastGenerator/index.php [hunk 2]
```diff
@@ -9,6 +9,17 @@
 ############################################################
 session_start();
 require 'core/include.php';
+
+// Backwards compatibility: Redirect pre-3.0 archive pages to
+// categories.php.
+if(isset($_GET['p'])) {
+    if($_GET['p'] == 'archive') {
+        $redirect_url = $config['url'] . 'categories.php?cat=' . $_GET['cat'];
+        header('Location: ' . $redirect_url, TRUE, 301);
```
Comment (emilengler) [inline PodcastGenerator/index.php]: Please remove the last two parameters. I know they are good but currently we don't use them at any function.
This is change of another PR.

Comment (cmpilato) [reply PodcastGenerator/index.php]: Done.

**Comment by emilengler** [comment]
Thanks, could you squash your commits?

**Comment by emilengler** [comment]
Thanks :)

**Pull request merged**